lib/spack/spack/cmd/mirror.py

```diff
@@ -226,11 +240,45 @@ def mirror_create(args):
             colify(s.cformat("$_$@") for s in error)
 
 
+def mirror_binary(args):
```

Comment:
This looks good mechanically, in that I think this is how files will need to be uploaded. Command line API-wise we need something higher level so that users don’t have to know how to structure a mirror.